src/pages/ShopingCart.page.js

```diff
@@ -22,4 +26,12 @@ export class ShopingCartPage extends BaseSwagLabPage {
     async removeCartItemById(id) {
         await this.cartItems.nth(id).locator(this.removeItemSelector).click();
     }
+
+    async getAllItemsInCart() {
```

Comment:
This looks not perfect yet. Everyone should know we should call this function before accessing this.allItemsInCart, but its not obvious. Please recheck on pattern, it could help. [Link](https://javascriptpatterns.vercel.app/patterns/design-patterns/proxy-pattern)